packages/cli/lib/zeroYaml/compile.ts

```diff
@@ -478,6 +478,71 @@ function nangoPlugin({ entryPoint }: { entryPoint: string }) {
                         }
                     },
 
+                    ExportNamedDeclaration(astPath) {
+                        // Skip internally transformed exports
+                        if ((astPath.node as any).__transformedByRemoveCreateWrappers) {
+                            return;
+                        }
+
+                        // Skip processing if the current file is not an entry point
+                        const currentFilePath = (astPath.hub as any)?.file?.opts?.filename;
+                        if (currentFilePath) {
+                            const normalizedCurrentPath = path.resolve(currentFilePath.replace('.ts', '.js'));
+                            if (normalizedCurrentPath !== realEntryPoint) {
+                                return;
+                            }
+                        }
+
+                        const lineNumber = astPath.node.loc?.start.line || 0;
+                        const node = astPath.node;
+
+                        // Check if this is a re-export (export { something } from 'module')
+                        if (node.source) {
+                            return; // Allow re-exports
+                        }
+
+                        // Check if any exported specifiers are not in allowedExports
+                        if (node.specifiers) {
+                            for (const specifier of node.specifiers) {
+                                if (t.isExportSpecifier(specifier) && t.isIdentifier(specifier.exported)) {
+                                    const exportedName = specifier.exported.name;
+                                    if (!allowedExports.includes(exportedName)) {
+                                        throw new CompileError(
+                                            'nango_named_export_not_allowed',
+                                            lineNumber,
+                                            `Named export '${exportedName}' is not allowed. Only export default and ${allowedExports.join(', ')} are permitted.`
+                                        );
+                                    }
+                                }
+                            }
```

Comment:
can we return here. if `node.specifier`, can it ever enter the if `node.declaratoin` below?